staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -846,7 +846,7 @@ func structToUnstructured(sv, dv reflect.Value) error {
 		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
 			realMap[fieldInfo.name] = fv.Int()
 		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-			realMap[fieldInfo.name] = fv.Uint()
+			realMap[fieldInfo.name] = int64(fv.Uint())
```

Comment:
This will overflow uint64 values > max int64 ... I'm not sure whether it is better to error or cast to a float64 in that case, but silently overflowing and going negative is definitely not what we want to do. Since going out to json and then back in would produce a float64, casting uint64 values > max int64 to float64 is probably better.

This should also add tests for the overflowing and non-overflowing cases to exercise the expected behavior.

Reply:
> realMap[fieldInfo.name] = fv.Uint()

I think we never want `uint` or `uint64` or any other unsigned type. We only want `int64` or `float64`, right?